internal/bminventory/inventory.go

```diff
@@ -1635,17 +1660,51 @@ func (b *bareMetalInventory) DisableHost(ctx context.Context, params installer.D
 		return common.GenerateErrorResponder(common.NewApiError(http.StatusInternalServerError, err))
 	}
 
+	if err := b.hostApi.RefreshStatus(ctx, &host, tx); err != nil {
+		msg := "Failed to refresh host status"
+		b.eventsHandler.AddEvent(ctx, params.ClusterID, &params.HostID, models.EventSeverityError, msg, time.Now())
+		return common.GenerateErrorResponder(common.NewApiError(http.StatusInternalServerError, err))
```

Comment:
I think that you can use only `common.NewApiError` in those cases` no need to have both `common.GenerateErrorResponder`  and `common.NewApiError`